component/common/loki/client/internal/marker_handler_test.go

```diff
@@ -0,0 +1,45 @@
+package internal
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+type mockMarkerFileHandler struct {
+	lastMarkedSegment int
+}
+
+func (m *mockMarkerFileHandler) LastMarkedSegment() int {
+	return m.lastMarkedSegment
+}
+
+func (m *mockMarkerFileHandler) MarkSegment(segment int) {
+	m.lastMarkedSegment = segment
+}
+
+func TestMarkerHandler(t *testing.T) {
```

Comment:
It'd be nice to add some fail-safety for corrupted marker files and also to have tests to check it.

Reply:
Yes, I'll add this once we agree on how the marker file should look like 😀